tools-v2/pkg/cli/command/curvebs/check/server/server.go

```diff
@@ -0,0 +1,237 @@
+	copysetbs "github.com/opencurve/curve/tools-v2/pkg/cli/command/curvebs/status/copyset"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/copyset"
+	"github.com/opencurve/curve/tools-v2/proto/proto/topology"
+	"github.com/spf13/cobra"
+)
+
+const (
+	serverExample = `$ curve bs check server`
+)
+
+type ServerCommand struct {
+	basecmd.FinalCurveCmd
+	ServerID uint32
+	ServerIP string
+	Port     uint32
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*ServerCommand)(nil)
+
+func NewServerCommand() *cobra.Command {
+	return NewCheckServerCommand().Cmd
+}
+
+func (sCmd *ServerCommand) AddFlags() {
+	config.AddBsMdsFlagOption(sCmd.Cmd)
+	config.AddRpcRetryTimesFlag(sCmd.Cmd)
+	config.AddRpcTimeoutFlag(sCmd.Cmd)
+	config.AddBsServerIdOptionFlag(sCmd.Cmd)
+	config.AddBsIpOptionFlag(sCmd.Cmd)
+	config.AddBsPortOptionFlag(sCmd.Cmd)
+}
+
+func NewCheckServerCommand() *ServerCommand {
+	ckCmd := &ServerCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "server",
+			Short:   "check all copysets infomation on server",
+			Example: serverExample,
+		},
+	}
+
+	basecmd.NewFinalCurveCli(&ckCmd.FinalCurveCmd, ckCmd)
+	return ckCmd
+}
+
+func (sCmd *ServerCommand) Init(cmd *cobra.Command, args []string) error {
+
+	serverID := config.GetBsFlagUint32(sCmd.Cmd, config.CURVEBS_SERVER_ID)
+	serverIP := config.GetBsFlagString(sCmd.Cmd, config.CURVEBS_IP)
+	port := config.GetBsFlagUint32(sCmd.Cmd, config.CURVEBS_PORT)
+	sCmd.ServerID = serverID
+	sCmd.ServerIP = serverIP
+	sCmd.Port = port
+
+	header := []string{cobrautil.ROW_SERVER, cobrautil.ROW_IP, cobrautil.ROW_TOTAL,
+		cobrautil.ROW_UNHEALTHY_COPYSET,
+	}
+	sCmd.SetHeader(header)
+	return nil
+}
+
+func (sCmd *ServerCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&sCmd.FinalCurveCmd, sCmd)
+}
+
+func (sCmd *ServerCommand) ResultPlainOutput() error {
+	return output.FinalCmdOutputPlain(&sCmd.FinalCurveCmd)
+}
+
+func (sCmd *ServerCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	// 1 send RPCs to get all ChunkServerInfo on server
+	timeout := config.GetFlagDuration(sCmd.Cmd, config.RPCTIMEOUT)
+	retrytimes := config.GetFlagInt32(sCmd.Cmd, config.RPCRETRYTIMES)
+	mdsAddrs, err := config.GetBsMdsAddrSlice(sCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+
+	var rpc *listchunkserver.ListChunkServerRpc
+	if sCmd.ServerIP != "" {
+		rpc = &listchunkserver.ListChunkServerRpc{
+			Request: &topology.ListChunkServerRequest{
+				Ip:   &sCmd.ServerIP,
+				Port: &sCmd.Port,
+			},
+			Info: basecmd.NewRpc(mdsAddrs, timeout, retrytimes, "ListChunkServer"),
+		}
+	} else {
+		rpc = &listchunkserver.ListChunkServerRpc{
+			Request: &topology.ListChunkServerRequest{
+				ServerID: &sCmd.ServerID,
+			},
+			Info: basecmd.NewRpc(mdsAddrs, timeout, retrytimes, "ListChunkServer"),
+		}
+	}
+
```

Comment:
Here you can separate the list chunkserver rpc into a separate command, and then call this command. For example [fileinfo.go](https://github.com/opencurve/curve/blob/master/tools-v2/pkg/cli/command/curvebs/query/file/fileinfo.go)
These things can be processed in the command.